packages/synapse-react-client/src/components/ChallengeTeamWizard/CreateChallengeTeam.tsx

```diff
@@ -1,101 +1,191 @@
+  'Please limit the initial number of invited members to three. You may add additional members after the team has been created.'
 
-export const CreateChallengeTeam = ({
-  onChangeTeamInfo,
-  onError,
-}: CreateChallengeTeamProps) => {
-  const [team, setTeam] = useState<CreateTeamRequest>({
-    name: '',
-    description: '',
-    message: '',
-    invitees: '',
-  })
+export type CreateChallengeTeamHandle = {
+  // Allow the parent component to trigger a submit of the form, so this may be embedded in an arbitrary modal.
+  submit: () => void
+}
 
-  const handleTeamUpdate = (update: Record<string, string>) => {
-    const updatedTeam: CreateTeamRequest = { ...team, ...update }
-    setTeam(updatedTeam)
-    onChangeTeamInfo(updatedTeam)
-  }
+export const CreateChallengeTeam = React.forwardRef(
+  function CreateChallengeTeam(
+    props: CreateChallengeTeamProps,
+    ref: React.ForwardedRef<CreateChallengeTeamHandle>,
+  ) {
+    const { challengeId, onCanSubmitChange = noop, onFinished = noop } = props
+    const [team, setTeam] = useState<CreateTeamRequest>({
+      name: '',
+      description: '',
+    })
+    const [invitationMessage, setInvitationMessage] = useState('')
+    const [rawInvitees, setRawInvitees] = useState('')
 
-  const handleInviteesUpdate = (inviteeString: string) => {
-    const invitees = inviteeString.split(',')
-    if (invitees.length > 3) {
-      if (onError)
-        return onError(
-          'Please limit the initial number of invited members to three. You may add additional members after the team has been created.',
-        )
+    const handleTeamUpdate = (update: Partial<Team>) => {
+      const updatedTeam: CreateTeamRequest = { ...team, ...update }
+      setTeam(updatedTeam)
     }
-    handleTeamUpdate({ invitees: inviteeString })
-  }
 
-  return (
-    <Box>
-      <Typography variant="body1" sx={{ lineHeight: '20px' }}>
-        Create a new team for this Challenge!
-      </Typography>
-      <TextField
-        id="name"
-        label="Team Name"
-        value={team.name}
-        fullWidth
-        autoFocus
-        required
-        onChange={event => handleTeamUpdate({ name: event.target.value })}
-      />
-      <Box display="flex">
+    const { inviteesParseResult, parsedInvitees } = useMemo(() => {
+      const inviteesParseResult = parse<string[]>(rawInvitees, {
+        delimiter: ',',
+        transform(value) {
+          return value.trim()
+        },
+      })
+      const parsedInvitees: string[] = inviteesParseResult.data[0] || []
+
+      return { inviteesParseResult, parsedInvitees }
+    }, [rawInvitees])
+
+    const numberOfInvitees = parsedInvitees.length
+    const tooManyInvitees = numberOfInvitees > INVITEE_LIMIT
+
+    const formDataIsValid = Boolean(
+      team && team.name && team.name.length > 1 && !tooManyInvitees,
+    )
+
+    useEffect(() => {
+      onCanSubmitChange(formDataIsValid)
+    }, [formDataIsValid, onCanSubmitChange])
```

Comment:
The parent will render the `Submit` button, so let the parent know when the form data becomes valid or invalid for submission so it can disable the button.